### PR TITLE
Use Activity Result APIs for External player

### PR DESCRIPTION
**Pull request by Maxr1998**
Squash merge please.

### CODE REVIEW

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt [hunk 1]
```diff
@@ -24,12 +25,36 @@ import org.koin.core.KoinComponent
...
+            Constants.MPV_PLAYER_RESULT_ACTION -> handleMPVPlayer(resultCode, intent)
+            Constants.MX_PLAYER_RESULT_ACTION -> handleMXPlayer(resultCode, intent)
+            Constants.VLC_PLAYER_RESULT_ACTION -> handleVLCPlayer(resultCode, intent)
+            else -> {
+                if (action != null && resultCode != Activity.RESULT_CANCELED) {
+                    Timber.d("Unknown action $action [resultCode: $resultCode]")
+                    notifyEvent(Constants.EVENT_CANCELED)
+                    context.toast(R.string.external_player_not_supported_yet, Toast.LENGTH_LONG)
+                } else {
+                    Timber.d("Playback canceled [no player selected or player without action result]")
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt:50]:
```suggestion
                    Timber.d("Playback cancelled: no player selected or player without action result")
```

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt [hunk 2]
```diff
@@ -24,12 +25,36 @@ import org.koin.core.KoinComponent
...
+    private val playerContract = registry.register("externalplayer", lifecycleOwner, ActivityResultContracts.StartActivityForResult()) { result ->
+        val resultCode = result.resultCode
+        val intent = result.data
+        when (val action = intent?.action) {
+            Constants.MPV_PLAYER_RESULT_ACTION -> handleMPVPlayer(resultCode, intent)
+            Constants.MX_PLAYER_RESULT_ACTION -> handleMXPlayer(resultCode, intent)
+            Constants.VLC_PLAYER_RESULT_ACTION -> handleVLCPlayer(resultCode, intent)
+            else -> {
+                if (action != null && resultCode != Activity.RESULT_CANCELED) {
+                    Timber.d("Unknown action $action [resultCode: $resultCode]")
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt:46]:
```suggestion
                    Timber.d("Unknown action $action (resultCode: $resultCode)")
```

**Review by nielsvanvelzen — APPROVED** [review]

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt (hunk 2 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt]: I actually prefer the square brackets here..

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt (hunk 1 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt]: Fixed.

**Review by nielsvanvelzen — APPROVED** [review]

**Review by thornbill — APPROVED** [review]

**Pull request merged**